proto/regen/ecocredit/v1/tx.proto

```diff
@@ -305,3 +326,95 @@ message MsgUpdateClassMetadata {
 
 // MsgUpdateClassMetadataResponse is the MsgUpdateClassMetadata response type.
 message MsgUpdateClassMetadataResponse {}
+
+// MsgNewCreditTypeRequest is the Msg/NewCreditTypeRequest request type.
+// The credit types are immutable and cannot be deleted. TODO(Tyler): don't think we want to allow updating? that could break stuff??
+message MsgNewCreditTypeRequest {
+  // credit_types is a list of new credit types to add
+  repeated CreditType credit_types = 1;
+
+  // root_address is the address of the caller.
+  // this MUST equal the address of the gov module for the tx to succeed.
+  string root_address = 2;
+}
+
+// MsgNewCreditTypeRequest is the Msg/NewCreditTypeResponse response type.
+message MsgNewCreditTypeResponse {}
+
+
+// MsgToggleAllowListRequest is the Msg/ToggleAllowListRequest request type
+message MsgToggleAllowListRequest {
+  // root_address is the address of the caller.
+  // this MUST equal the address of the gov module for the tx to succeed.
+  string root_address = 1;
+
+  // toggle is what the allow list will be set to.
+  bool toggle = 2;
+}
+
+// MsgToggleAllowListResponse is the Msg/MsgToggleAllowListResponse response type.
+message MsgToggleAllowListResponse {}
+
+// MsgUpdateAllowedCreditClassCreatorsRequest is the Msg/UpdateAllowedCreditClassCreators request type
+message MsgUpdateAllowedCreditClassCreatorsRequest {
+  // root_address is the address of the caller.
+  // this MUST equal the address of the gov module for the tx to succeed.
+  string root_address = 1;
+
+  // add_creators is a list of addresses to be added to the allow list for credit class creation.
+  repeated string add_creators = 2;
+
+  // remove_creators is a list of addresses to be removed from the allow list for credit class creation.
+  repeated string remove_creators = 3;
+}
+
+// MsgNewCreditClassCreatorResponse is the Msg/UpdateAllowedCreditClassCreators response type
+message MsgUpdateAllowedCreditClassCreatorsResponse {}
+
+// MsgUpdateCreditClassFeeRequest is the Msg/UpdateCreditClassFee request type
+message MsgUpdateCreditClassFeeRequest {
+  // root_address is the address of the caller.
+  // this MUST equal the address of the gov module for the tx to succeed.
+  string root_address = 1;
+
+  // add_fees are the coin denoms and amounts to be added to the allowed credit class creation fees.
+  repeated Fee add_fees = 2;
+
+  // remove_fees are the coin denoms and amounts to be removed from the allowed credit class creation fees.
+  repeated Fee remove_fees = 3;
+
+  // Fee represents a fee to be used in the credit class creation fee.
+  message Fee {
+    // denom is the coin denom
+    string denom = 1;
+    // amount is the amount of the coin needed to create the credit class.
+    string amount = 2;
+  }
+}
```

Comment:
This is a new feature that we might need to further discuss on a product call. I think for now we want to have REGEN be the only token accepted for fees on our network.